pkg/operator/operator.go

```diff
@@ -136,6 +136,11 @@ func NewOperator() (context.Context, *Operator) {
 		return []string{o.(*v1alpha5.Machine).Status.ProviderID}
 	}), "failed to setup machine provider id indexer")
 
+	lo.Must0(mgr.AddReadyzCheck("manager", func(req *http.Request) error {
+		mgr.GetCache().WaitForCacheSync(req.Context())
```

Comment:
Does this hold the thread until the cache is synced?

Reply:
If so, you probably want to do the check `IsCacheSynced` rather than `WaitForCacheSynced`

Reply:
It does hang, but hanging is caught by the readiness probe's client timeout.